### ISSUE TITLE
Ideas from lc0

### ISSUE THREAD

**Issue by TFiFiE** [leela-zero/leela-zero#2456]
https://medium.com/@veedrac/leela-chess-test40-test50-and-beyond-c15896becfac

This is a somewhat old article giving an overview of a few ideas proposed for or actually used in Lc0, but it hasn't been mentioned here before.

Especially the idea of managing time based on how much a visit distribution has changed over time seems nifty.

**Comment by tapsika** [comment]
I think the most important idea is simply to start doing new things and experiments.

To exaggerate a bit LZ's current slogan is "nothing ever happens". The idea of simply copying AGZ and everything will be fine is attractive, but this won't necessarily work out in practice (which may be illustrated by recent stall of progress). A perfect copy is hard because of all the unknown details and differences.

LC0 is (about to) try kldgain while AFAIK LZ haven't even done the most basic experiments:

Selfplay visits were reduced to 1600 to speed up game generation, and further (effectively) reduced by batching. It was never verified if these reductions increase learning speed or not. Running selfplay at varying visits was another attractive idea.

Puct / exploration: I'm not sure but guess these were adjusted to maximal playing strength. But was it ever verified to be optimal for selfplay as well? (Eg. slightly different settings may be slightly weaker but significantly more effective teaching new things.)

Even without having to code anything a lot of interesting tests could (have been) run.

**Comment by Friday9i** [comment]
Yeah, but "nothing ever happens" for LZ, and I'm afraid it won't improve in the future...

Good news is, KataGo exists :-)))))).
And KataGo is almost as strong as LZ (and will probably be much better in the coming months). And more than that, it includes so many new things and improvements: good management of handicap, adjustable komi, different board sizes (from 9x9 to 37x37, but smaller boards, ie 5x5 are also supported), non-square board sizes (eg 9x19!), alternative rule sets, non-slack moves when leading, no ladder issues, etc...
And KataGo is also progressing so fast, that just amazing! It reached ~LZ200 level at 1600 visits despite a smaller 20b net and in only 19 days where LZ took 18 months with a comparable computing power...

**Comment by tapsika** [comment]
Katago is attractive, but I'm not sure how it compares strength-wise without the ladder bonus. It may take 1-2 more runs to really take over LZ.

And a pure zero bot is still interesting, something like a reference point. So in any case I hope LZ will reach AGZ level sometime. But some optimum testing and maybe a few adjustments could prove useful for this (in terms of time or games required).

**Comment by l1t1** [comment]
https://github.com/LeelaChessZero/lc0/wiki/Third-Party-Nets is a good place to get weights
